presto-main/src/main/java/com/facebook/presto/sql/planner/QueryPlanner.java

```diff
@@ -281,10 +279,13 @@ private PlanBuilder planFrom(QuerySpecification node)
 
     private PlanBuilder planBuilderFor(PlanBuilder builder, Scope scope, Iterable<? extends Expression> expressionsToRemap)
     {
-        Map<Expression, VariableReferenceExpression> expressionsToVariables = variablesForExpressions(builder, expressionsToRemap);
         PlanBuilder newBuilder = planBuilderFor(builder, scope);
-        expressionsToVariables.entrySet()
-                .forEach(entry -> newBuilder.getTranslations().put(entry.getKey(), entry.getValue()));
+        for (Expression expression : expressionsToRemap) {
```

Comment:
Maybe just do
```
        stream(expressionsToRemap)
                .forEach(expression -> newBuilder.getTranslations().put(expression, builder.translate(expression)));
```
I don't have strong opinions. Just that overall the codebase seems to prefer stream API when logic is simple.